app/Jobs/CreateCallPowerPostInRogue.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Chompy\Jobs;
+
+use Exception;
+use Chompy\Services\Rogue;
+use Illuminate\Http\Request;
+use Illuminate\Bus\Queueable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+
+
+class CreateCallPowerPostInRogue implements ShouldQueue
+{
+	use Dispatchable, InteractsWithQueue, Queueable;
+
+	/**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+	public function __construct()
+	{
+	}
+
+    /**
+     * Execute the job to create a CallPower post in Rogue.
+     *
+     * @return array
+     */
+    public function handle(Rogue $rogue, Request $request)
+    {
+		// Using the mobile number, get or create a northstar_id.
+    	$user = $this->getOrCreateUser($request['mobile']);
+
+        // Using the callpower_campaign_id, get the action_id from Rogue.
+    	$action = $rogue->getActionFromCallPowerCampaignId($request['callpower_campaign_id']);
+
+        // Check if the post exists in Rogue. If not, create the post.
+    	$existingPost = $rogue->getPost([
+    		'action_id' => $action['data'][0]['id'],
```

Comment:
It seems a little strange to have to grab this nested `['data'][0]` path from a variable named "action". I think it would be a little cleaner to parse out the action itself in `getActionFromCallPowerCampaignId`, so we don't have to remember the JSON format each time we use this.

Reply:
ah yes way better!